pkg/cloud/aws/tags/types.go

```diff
@@ -26,6 +29,24 @@ func (m Map) Equals(other Map) bool {
 	return reflect.DeepEqual(m, other)
 }
 
+// HasOwned returns true if the tags contains a tag that marks the resource as owned by the cluster.
+func (m Map) HasOwned(cluster string) bool {
```

Comment:
That's gonna be an interesting use case, when the VPC is shared between clusters, but that's probably out of scope for v1alpha1.

Reply:
I'm not sure we should consider it out of scope. In the case of separation of concerns users might very well choose to re-use the same VPC to avoid having to request a VPC from another team for each cluster they create.